### PR TITLE
Custom rules

### PR DESCRIPTION
**Pull request by vojty**
- [x] split JS into 2 repositories - rules & config
- [x] create eslint-plugin-gooddata and add it to config
- [ ] markdown explanation for the rules?
- [ ] release

### CODE REVIEW

File: javascript/src/rules/no-lodash-bindall.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = (context) => {
+    return {
+        ExpressionStatement: (node) => {
+            if (node.expression.callee.name === 'bindAll') {
+                context.report(node, 'bindAll is forbidden (no function names in string). Use this.myFunction.bind(this).');
```
Comment (crudo) [inline javascript/src/rules/no-lodash-bindall.js]: 👍

Comment (vojty) [reply javascript/src/rules/no-lodash-bindall.js]: any other nazi ideas?

File: javascript/rules/src/rules/tests/no-object-assign.spec.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            experimentalObjectRestSpread: true,
+            jsx: true
+        },
+        sourceType: 'module'
+    }
+});
+
+const variableDeclarationErrors = [
+    {
+        message: 'Object.assign is forbidden, use destructuring assignment instead.',
```
Comment (vaclavbohac) [inline javascript/rules/src/rules/tests/no-object-assign.spec.js]: `use object spread operator instead`?

File: javascript/rules/src/rules/tests/no-object-assign.spec.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        type: 'VariableDeclaration',
+    }
+];
+
+const expressionStatementErrors = [
+    {
+        message: 'Object.assign is forbidden, use destructuring assignment instead.',
```
Comment (vaclavbohac) [inline javascript/rules/src/rules/tests/no-object-assign.spec.js]: `use object spread operator instead`?

File: javascript/rules/src/rules/tests/no-object-assign.spec.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        type: 'ExpressionStatement',
+    }
+];
+
+const returnStatementErrors = [
+    {
+        message: 'Object.assign is forbidden, use destructuring assignment instead.',
```
Comment (vaclavbohac) [inline javascript/rules/src/rules/tests/no-object-assign.spec.js]: `use object spread operator instead`?

File: javascript/rules/src/rules/no-object-assign.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        }
+
+        if (callee.object && callee.object.name === 'Object' && callee.property && callee.property.name === 'assign') {
+            return true;
+        }
+        return false;
+    }
+
+    function raiseError(node) {
+        context.report(node, 'Object.assign is forbidden, use destructuring assignment instead.');
```
Comment (vaclavbohac) [inline javascript/rules/src/rules/no-object-assign.js]: `use object spread operator instead`?

File: javascript/rules/src/rules/tests/no-react-addons-test-utils.spec.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    env: {
+        es6: true
+    },
+    parserOptions: {
+        sourceType: 'module'
+    }
+});
+
+const errors = [{
+    message: 'Use Enyzme instead of react-addons-test-utils.',
```
Comment (vaclavbohac) [inline javascript/rules/src/rules/tests/no-react-addons-test-utils.spec.js]: `Enzyme`

File: javascript/rules/src/rules/no-react-addons-test-utils.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+module.exports = (context) => {
+    return {
+        ImportDeclaration: (node) => {
+            if (node.source.value === 'react-addons-test-utils') {
+                context.report(node, 'Use Enyzme instead of react-addons-test-utils.');
```
Comment (vaclavbohac) [inline javascript/rules/src/rules/no-react-addons-test-utils.js]: `Enzyme`

**Pull request closed without merging**